File: LogAndroid/BaseLib/src/main/kotlin/jp/co/toukei/log/lib/util/ReuseDateFormatter.kt
package jp.co.toukei.log.lib.util

import java.text.ParsePosition
import java.text.SimpleDateFormat
import java.util.*

class ReuseDateFormatter(
        private val pattern: String,
        private val locale: Locale,
        private val timeZone: TimeZone
) : ReuseObject<SimpleDateFormat>() {

    override fun make(): SimpleDateFormat {
        return SimpleDateFormat(pattern, locale).also { it.timeZone = timeZone }
    }

    fun format(date: Date): String = format(date.time)
    fun format(date: Long): String = use { it.format(date) }
    fun parse(text: String): Date? = use { it.parse(text, ParsePosition(0)) ?: null }
}

File: LogiOS/TrustarApp/Ext/ReuseDateFormatter.swift
//
//  ReuseDateFormatter.swift
//  TrustarApp
//
//  Created by DION COMPANY on 2022/02/16.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation
class ReuseDateFormatter {
    private var pattern: String = "yyyy-MM-DD  HH:mm:ss"
    private var locale: Locale = Locale(identifier: "ja_JP")
    private var timeZone: TimeZone? = TimeZone(identifier: "Asia/Tokyo")
    private var dateFormatter: DateFormatter  = DateFormatter()
    
    init (pattern: String, locale: Locale, timeZone: TimeZone?) {
        self.pattern = pattern
        self.locale = locale
        self.timeZone = timeZone

        self.dateFormatter.dateFormat = self.pattern
        self.dateFormatter.timeZone = self.timeZone
        self.dateFormatter.locale = self.locale
    }
    
    func format(date: Date) -> String {
        return dateFormatter.string(from: date)
    }
    
    func format(date: Int64) -> String {
        let dateVal = Date(miliseconds:  date)
        return dateFormatter.string(from: dateVal)
    }
    
    func parse(text: String) -> Date? {
        return dateFormatter.date(from: text)
    }
}
